src/index.ts

```diff
@@ -12,43 +22,46 @@ function errorHandler(error: any, request: FastifyRequest, reply: FastifyReply<S
     request.log.error({ ...error, statusCode })
   }
 
-  const response = isClientError
-    ? {
-        error: error.type,
-        message: error.message,
-        statusCode,
-        details: error.validation || error.details,
-      }
-    : {
-        error: error.type,
-        message: 'Something went wrong. Try again later.',
-        statusCode,
-      }
+  let response: any
+  if (isInterference(error) && isClientError) {
+    response = {
+      error: error.type,
+      message: error.message,
+      details: error.details,
+      statusCode,
+    }
+  } else if (error.validation) {
+    response = {
+      error: 'INVALID_REQUEST',
+      message: error.message,
+      details: error.validation,
+      statusCode,
+    }
+  } else {
+    response = {
+      error: 'INTERNAL_ERROR',
+      message: 'An error occurred',
+      details: {},
+      statusCode,
+    }
+  }
 
   reply.code(statusCode).send(response)
 }
 
-interface Codes {
-  [key: string]: number
-}
-
-export interface InterferencePluginOptions {
-  codes?: Codes
-}
-
 export default fp(
-  async function interferencePlugin(fastify, opts: InterferencePluginOptions) {
-    fastify.setErrorHandler(errorHandler)
-    fastify.decorateReply('httpCodes', opts.codes ?? {})
+  async function interferencePlugin(fastify) {
+    fastify.decorate('interference', new Map<string, number>())
+    fastify.setErrorHandler(errorHandler.bind(fastify))
```

Comment:
This bind is not necessary since Fastify internally already do it.

Reply:
Sadly, I found a regression about that bind. fastify@2.14.1

Reply:
We can leave it as is for now. I'll double check on Fastify's repository and patch if needed